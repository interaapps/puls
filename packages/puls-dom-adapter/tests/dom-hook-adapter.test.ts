import {createTemplateFunction} from "pulsjs-template";
import {PulsHookedDOMAdapter} from "../index";
import {state} from "pulsjs-state";

const html = createTemplateFunction()
test('dom-hook-adapter-test', () => {
    const d = new PulsHookedDOMAdapter(html`test`.parse())

    const [el] = d.render()
    expect(el.textContent).toMatchSnapshot()
})

test('dom-hook-adapter-full', () => {
    const name = state('test')

    const d = new PulsHookedDOMAdapter(html`
        <div>
            <h1>Hello, World!</h1>
            <p>${name}</p>
        </div>
    `.parse())
    
    const [div] = d.render()

    name.value = 'John'

    expect((div as HTMLElement).innerHTML).toMatchSnapshot()
})