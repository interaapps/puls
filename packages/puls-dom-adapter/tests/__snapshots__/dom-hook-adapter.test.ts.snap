// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dom-hook-adapter-full 1`] = `"<h1>Hello, World!</h1><p>John</p>"`;

exports[`dom-hook-adapter-test 1`] = `"test"`;
